exercises/Toluwalase_Agoro/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,184 @@
+		{
+			stringName.erase(i, 1);
+			i--;
+		}
+		i++;
+	}
+}
+
+
+string getWord(ifstream &stream) {
+	/* Function accept an input stream object and return a string from 
+	input stream by reading stream 1 character at a time and concatenating
+	it to a string until a space ,newline or '-'  character.
+	*/
+	string OutputString{};
+	char InputCharacter{};
+
+	while (true) {
+		stream.get(InputCharacter);
+		if (InputCharacter == '-' || InputCharacter == ' ' || InputCharacter == '\n'||stream.eof())
+			break;
+		OutputString += InputCharacter;
+	}
+	return OutputString;
+}
+
+
+int findString(const string &SearchString, const vector <string> &SearchList) {
+	/*Function accept a string and vector of strings and searches for given string in 
+	vector returns index at which it is found or -1 if it does not exist 
+	in vector
+	*/
+	for (int Index = 0; Index < SearchList.size(); Index++) {
+		if (SearchList[Index] == SearchString) {
+			return Index;
+		}
+	}
+	return -1;
+}
+
+
+int findInt(const int &SearchInteger,const vector <int> &SearchList) {
+	/*Function accept an integer and vector of integers and searches for given integer in
+	vector returns index at which it is found or -1 if it does not exist
+	in vector
+	*/
+	for (int Index = 0; Index < SearchList.size(); Index++) {
+		if (SearchList[Index] == SearchInteger) {
+			return Index;
+		}
+	}
+	return -1;
+}
+
+
+int main() { 
+	/*This block of code declares all vectors used 
+	and gets input from user so a to open 
+	input file
+	*/
+	vector <string> VectorOfWords{};
+	vector <int> VectorOfCounts{};
+	vector <int> VectorOfOrderToRead{};
+	char FilePathToInput [100];
+	cout << "Please input a file path to input file :";
+	cin.getline(FilePathToInput,100);
+	ifstream FileToRead;
+	FileToRead.open(FilePathToInput);
+	
+	if (!FileToRead) {
+		/* This if block checks if file was opened correctly 
+		and if it was not program is terminated
+		*/
+		cout << "File could not be opened" << endl;
+		cout << "Ensure that file path is valid" << endl;
+		return 1;
+	}
+	while(!FileToRead.eof()) {
+		/* This while block reads input file and adds formatted words to
+		VectorOfWords until the end of file has been reached. This block 
+		has an if , else if block that determines whether word is more than 
+		4 letters and already exists in VectorOfWords. if it does it increments 
+		the count in VectorOfCounts otherwise it is added to VectorOfWords
+		*/
+		string HolderString{ };
+		HolderString = getWord(FileToRead);
+		toLower(HolderString);
+		removePunctuation(HolderString);
+		int location = findString(HolderString, VectorOfWords);
+		if (HolderString.size() > 4 && location == -1) {
+			VectorOfWords.push_back(HolderString);
+			VectorOfCounts.push_back(1);
+		}
+		else {
+			if (HolderString.size() > 4 && location != -1)
+			{
+				VectorOfCounts.at(location) += 1;
+		}
+		}
+	}
+	FileToRead.close();
+
+	for (int i = 0; i < VectorOfWords.size(); i++) {
```

Comment:
I quite like what you've done here in many ways in that it's quite a novel way of generating the sorted list of counts. However I'm going to ask you to change it for 2 reasons:
a) the double loop is quite slow because the two vectors you're looping over are ~1500 elements each
b) there are already standard functions in the standard library for sorting vectors that mean you don't have to expend as much thought on this part
Have a looking at the standard library "sort" function. I think to use it here you will perhaps need to reformulate the VectorOfWords and VectorOfCounts into a single vector containing pairs